src/crypto/cmac.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @fileoverview This module implements AES-CMAC on top of
+ * native AES-CBC using either the WebCrypto API or Node.js' crypto API.
+ * @requires asmcrypto.js
+ * @requires util
+ * @module crypto/cmac
+ */
+
+import { AES_CBC } from 'asmcrypto.js/src/aes/cbc/exports';
+import util from '../util';
+
+const webCrypto = util.getWebCryptoAll();
+const nodeCrypto = util.getNodeCrypto();
+const Buffer = util.getNodeBuffer();
+
+
+/**
+ * This implementation of CMAC is based on the description of OMAC in
+ * http://web.cs.ucdavis.edu/~rogaway/papers/eax.pdf. As per that
+ * document:
+ *
+ * We have made a small modification to the OMAC algorithm as it was
+ * originally presented, changing one of its two constants.
+ * Specifically, the constant 4 at line 40 was the constant 1/2 (the
+ * multiplicative inverse of 2) in the original definition of OMAC [14].
+ * The OMAC authors indicate that they will promulgate this modification
+ * [15], which slightly simplifies implementations.
+ */
+
+const blockLength = 16;
+
+
+/**
+ * xor `padding` into the end of `data`. This function implements "the
+ * operation xor→ [which] xors the shorter string into the end of longer
+ * one". Since data is always as least as long as padding, we can
+ * simplify the implementation.
+ * @param {Uint8Array} data
+ * @param {Uint8Array} padding
+ */
+function rXorMut(data, padding) {
```

Comment:
this function makes sense to me, but not its name :D 
I can see the variable r in the paper, but could we name this something more generic?

Reply:
I meant for it to stand for "Right Xor which Mutates its argument". I guess I should spell out the r. I noticed that some other functions just mutate their argument without telling anyone, but it made sense to me to keep the distinction because e.g. eax.js has both `xor` and `xorMut`, currently. We could just replace the former with `xor(a.slice(), b)` though, and call this function `rightXor`, if you prefer? Since JS vars support unicode, we could call it `xorǃ` :smiling_imp:

Reply:
I am fine with either rightXor or rightXorMut, I think the "r" was the part that confused me :)